### PR TITLE
ramips: add support for ASUS RT-AX53U

### PR DESCRIPTION
**Pull request by ccchen1984**
Specifications:
- Device: ASUS RT-AX53U
- SoC: MT7621AT
- Flash: 128MB
- RAM: 256MB
- Switch: 1 WAN, 3 LAN (10/100/1000 Mbps)
- WiFi: MT7905 2x2 2.4G + MT7975 2x2 5G
- Ports: USB 3.0
- LEDs: 1x POWER (blue, configurable)
        3x LAN (blue, configurable)
        1x WAN (blue, configurable)
        1x USB (blue, not configurable)
	1x 2.4G (blue, not configurable)
	1x 5G (blue, not configurable)

Flash by U-Boot TFTP method:
- Configure your PC with IP 192.168.1.2
- Set up TFTP server and put the factory.bin image on your PC
- Connect serial port(rate:115200) and turn on AP, then interrupt "U-Boot Boot Menu" by hitting any key
   Select "2. Upgrade firmware"
   Press enter when show "Run firmware after upgrading? (Y/n):"
   Select 0 for TFTP method
   Input U-Boot's IP address: 192.168.1.1
   Input TFTP server's IP address: 192.168.1.2
   Input IP netmask: 255.255.255.0
   Input file name: openwrt-ramips-mt7621-asus_rt-ax53u-squashfs-factory.bin
- Restart AP aftre see the log "Firmware upgrade completed!"

Signed-off-by: Chuncheng Chen <ccchen1984@gmail.com>

### CODE REVIEW

**Comment by castiel652** [comment]
Should merge into master branch first

File: target/linux/ramips/dts/mt7621_asus_rt-ax53u.dts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+&factory {
+	compatible = "nvmem-cells";
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	macaddr_factory_4: macaddr@4 {
+		reg = <0x4 0x6>;
+	};
+};
```
Comment (hauke) [inline target/linux/ramips/dts/mt7621_asus_rt-ax53u.dts:162]: Please add an empty line at the end of the file.

File: target/linux/ramips/image/mt7621.mk [hunk 2]
```diff
@@ -227,6 +227,23 @@ define Device/asus_rt-n56u-b1
...
+  IMAGE_SIZE := 51200k
+  UBINIZE_OPTS := -E 5
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE := 4096k
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | \
+	check-size
+  DEVICE_PACKAGES :=  kmod-mt7915e kmod-usb3 uboot-envtools
```
Comment (hauke) [inline target/linux/ramips/image/mt7621.mk:243]: If you add uboot-entools you should also add it to the package/boot/uboot-envtools/files/ramips file.

`fw_printenv` should then show your u-boot environment.

**Comment by hauke** [comment]
Please rebase this on top of master, currently it does not apply on top of master.

File: target/linux/ramips/dts/mt7621_asus_rt-ax53u.dts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0xe0000>;
+			read-only;
+		};
+
+		partition@e0000 {
+			label = "u-boot-env";
+			reg = <0xe0000 0x100000>;
+			read-only;
```
Comment (csharper2005) [inline target/linux/ramips/dts/mt7621_asus_rt-ax53u.dts:69]: Why not to make it writable?

Comment (ccchen1984) [reply target/linux/ramips/dts/mt7621_asus_rt-ax53u.dts:69]: I will remove uboot-envtools later

**Comment by ynezz** [comment]
Superseded by #10448

**Pull request closed without merging**